sdk/src/Core/Amazon.Runtime/Credentials/DefaultAWSCredentialsIdentityResolver.cs

```diff
@@ -51,8 +52,8 @@ public DefaultAWSCredentialsIdentityResolver()
 #if BCL
                     () => new AppConfigAWSCredentials(), // Test explicit keys/profile name first.
 #endif
-                    () => AssumeRoleWithWebIdentityCredentials.FromEnvironmentVariables(),
                     () => new EnvironmentVariablesAWSCredentials(), // Look for credentials set in environment vars.
+                    () => AssumeRoleWithWebIdentityCredentials.FromEnvironmentVariables(),
```

Comment:
**Important**: This change is intentional! The search order now matches the `credentials-provider-chain-search-precedence` SEP. 

The order defined in the spec is: `Environment variables: credentials` -> `Environment variables: web identity token / STS` -> `Profile: credentials`.